ayon_api/server_api.py

```diff
@@ -8134,11 +8134,14 @@ def send_batch_operations(
         return op_results
 
     def _convert_entity_data(self, entity):
-        if not entity:
+        if not entity or "data" not in entity:
             return
-        entity_data = entity.get("data")
-        if (
-            entity_data is not None
-            and isinstance(entity_data, str)
-        ):
-            entity["data"] = json.loads(entity_data)
+
+        entity_data = entity["data"]
+        if entity_data is None:
+            entity_data = {}
```

Comment:
Isn't this quite similar as to just doing `entity_data = entity.get("data") or {}`

Reply:
`"data"` should not be filled if are not available in entity.

Reply:
I see - we may want to update the PR description then to explain that _with this PR_ it entity will also lack `"data"` completely whereas previously it may have been `None`. The PR isn't necessarily enforcing `data` to be `dict` but only enforcing that it will be - if it was existing in in the input data.